lib/backend/firestore/firestorebk.go

```diff
@@ -110,13 +110,63 @@ type FirestoreBackend struct {
 }
 
 type record struct {
+	Key       string `firestore:"key,omitempty"`
```

Comment:
Should `Key` be changed to `[]byte` as well while we're here?  I don't think we enforce UTF-8 on keys either.

Reply:
Good point. In practice it seems to always be UTF-8, but it's not enforced anywhere.
Changed and added test coverage for other backends.